src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ADD_OPTION = 'addProperties';
+  const REMOVE_OPTION = 'removeProperties';
+  const CLEAR_OPTION = 'clear';
+  const allState = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ADD_OPTION:
+        for (const key of Object.keys(action.extraData)) {
+          stateCopy[key] = action.extraData[key];
+        }
+
+        break;
+      case REMOVE_OPTION:
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+
+        break;
+      case CLEAR_OPTION:
+        for (const key of Object.keys(stateCopy)) {
+          delete stateCopy[key];
+        }
+
+        break;
+      default:
+        return 'Something wrong';
```

Comment:
this case should throw an error, but not return string.
[MDN about throw](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Statements/throw)
[MDN about error](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Error)